department-of-veterans-affairs/abd-vro
# EP Merge: Create end 2 end tests

## What was the problem?

Current integration tests are only truly integrated with RabbitMQ and Postgres.

Associated tickets or Slack threads:
- #2423
- Stacked on:
  - #2429 

## How does this fix it?[^1]

These updates add end to end tests that utilize RabbitMQ, postgres, `svc-bip-api`, and `svc-bgs-api` as well as their corresponding mock services to send new merge requests and test several cases:
- both pending claim ID and EP400 claim ids result in successful responses from mocks
- pending claim ID results in failure for getting claim details, and successfully removes special issues from EP400
- pending claim ID results in failure for getting claim details, and failure to remove special issues from EP400
- pending claim ID results in failure for getting claim contentions, and successfully removes special issues from EP400
- pending claim ID results in failure for getting claim contentions, and failure to remove special issues from EP400
- EP400 claim ID results in failure for getting claim contentions
- EP400 claim ID results in failure to set temporary station of jurisdiction
- pending claim ID results in failure to move (create) contentions in pending claim
- EP400 claim ID results in failure to cancel claim contentions 

## How to test this PR
- pull code
- Unit Test:
  - from /domain-ee/ee-ep-merge-app run pytest
- Integration Tests:
  - run [base platform](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Docker-Compose#platform-base)
  - from /domain-ee/ee-ep-merge-app run pytest ./integration
- End to End tests:
  - run this [CI](https://github.com/department-of-veterans-affairs/abd-vro/actions/workflows/ee-ep-merge-end-to-end.yml) action to verify successful integration test
  - run end to end tests locally:
    - run bgs and bip services and mocks:
      - `COMPOSE_PROFILES="bip,bgs" ./gradlew :app:dockerComposeUp`
      - `COMPOSE_PROFILES="bip,bgs" ./gradlew -p mocks :dockerComposeUp`
    - `./gradlew :domain-ee:ee-ep-merge-app:endToEndTest`



[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
